python/beeai_framework/backend/types.py

```diff
@@ -38,11 +37,6 @@ class ChatModelParameters(BaseModel):
     stream: bool | None = None
 
 
-class ChatConfig(BaseModel):
-    # TODO: cache: ChatModelCache | Callable[[ChatModelCache], ChatModelCache] | None = None
```

Comment:
Given we are in the process of adding cache don't we want to keep this since the intention is to also include cache in the config

Reply:
or do you mean to flatten it and add cache as another param on the function?

Reply:
@vabarbosa just created cache constructs, but they are not used anywhere. 
Caching for ChatModel must be implemented later on.

Reply:
IIUC now that #602 is merged he's starting on updating the code to use it, like here.

Reply:
It will take a while. I would merge this and then create a separate PR once done.

Reply:
I'm ok with this if the intention is to permanently remove `ChatConfig` and "flatten" the inputs in `config()`, but if that's the case I would ask that the commented out line here be added to the inputs there and commented out.

Reply:
Got it. Updated.